packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart

```diff
@@ -36,74 +40,144 @@ class MonitorVerbHandler extends AbstractVerbHandler {
+      if (notification.notification!.contains(RegExp(regex)) ||
+          (fromAtSign != null && fromAtSign.contains(RegExp(regex)))) {
+        atConnection
+            .write('notification: ${jsonEncode(notification.toJson())}\n');
+      }
+    } on FormatException {
+      logger.severe('Invalid regular expression : $regex');
+      throw InvalidSyntaxException(
+          'Invalid regular expression. $regex is not a valid regex');
+    }
+  }
 
-    // If monitor verb contains a regular expression,
-    // push only if the notification matches regex
-    if (regex != null) {
-      logger.finer('regex is not null:$regex');
-      logger.finer('key: $key');
-      try {
-        // if key matches the regular expression, push notification.
-        // else if fromAtSign matches the regular expression, push notification.
-        if (key!.contains(RegExp(regex!))) {
-          logger.finer('key matches regex');
-          atConnection
-              .write('notification: ${jsonEncode(notification.toJson())}\n');
-        } else if (fromAtSign != null && fromAtSign.contains(RegExp(regex!))) {
-          logger.finer('fromAtSign matches regex');
-          atConnection
-              .write('notification: ${jsonEncode(notification.toJson())}\n');
-        } else {
-          logger.finer('no regex match');
-        }
-      } on FormatException {
-        logger.severe('Invalid regular expression : $regex');
-        throw InvalidSyntaxException('Invalid regular expression syntax');
+  /// If connection is authenticated via APKAM
+  ///    - Writes the notifications only if the namespace in the notification key
+  ///      matches the namespace in the enrollment.
+  ///    - Optionally if regex is supplied, write only the notifications that
+  ///      matches the pattern.
+  Future<void> _sendNotificationByEnrollmentNamespaceAccess(
+      Notification notification) async {
+    // Fetch namespaces that are associated with the enrollmentId.
+    var enrollmentKey =
+        '${(atConnection.getMetaData() as InboundConnectionMetadata).enrollmentId}.$newEnrollmentKeyPattern.$enrollManageNamespace${AtSecondaryServerImpl.getInstance().currentAtSign}';
+    EnrollDataStoreValue enrollDataStoreValue =
+        await getEnrollDataStoreValue(enrollmentKey);
+    // When an enrollment is revoked, avoid sending notifications to the
+    // existing monitor connection.
+    if (enrollDataStoreValue.approval!.state != EnrollStatus.approved.name) {
+      logger.info('Enrollment is not approved. Failed to send notifications');
+      return;
+    }
+    if (enrollDataStoreValue.namespaces.isEmpty) {
+      logger.info('No namespaces are enrolled for the enrollmentId:'
+          ' ${(atConnection.getMetaData() as InboundConnectionMetadata).enrollmentId}');
+      return;
+    }
+    // separate namespace from the notification key
+    String namespaceFromKey = notification.notification!
+        .substring(notification.notification!.indexOf('.') + 1);
```

Comment:
Need to handle situation where there is no namespace

Reply:
Added a check before this line to check if the `notification.notification` contains ".". If not available, it indicates the namespace is not present and do nothing.